src/pudl/package_data/ferc1/dbf_to_xbrl.csv

```diff
@@ -3215,62 +3215,62 @@ f1_respdnt_control,24,1994,Line 24,,,,
-f1_retained_erng,8,1994,Credit:,,,,
-f1_retained_erng,9,1994,TOTAL Credits to Retained Earnings (Acct. 439),,,,
-f1_retained_erng,10,1994,Debit:,,,,
-f1_retained_erng,11,1994,Debit:,,,,
-f1_retained_erng,12,1994,Debit:,,,,
-f1_retained_erng,13,1994,Debit:,,,,
-f1_retained_erng,14,1994,Debit:,,,,
-f1_retained_erng,15,1994,TOTAL Debits to Retained Earnings (Acct. 439),,,,
-f1_retained_erng,16,1994,Balance Transferred from Income (Account 433 less Account 418.1),,,,
-f1_retained_erng,17,1994,Appropriations of Retained Earnings (Acct. 436),,,,
-f1_retained_erng,18,1994,Line 18,,,,
-f1_retained_erng,19,1994,Line 19,,,,
-f1_retained_erng,20,1994,Line 20,,,,
-f1_retained_erng,21,1994,Line 21,,,,
-f1_retained_erng,22,1994,TOTAL Appropriations of Retained Earnings (Acct. 436),,,,
-f1_retained_erng,23,1994,Dividends Declared-Preferred Stock (Account 437),,,,
-f1_retained_erng,24,1994,Line 24,,,,
-f1_retained_erng,25,1994,Line 25,,,,
-f1_retained_erng,26,1994,Line 26,,,,
-f1_retained_erng,27,1994,Line 27,,,,
-f1_retained_erng,28,1994,Line 28,,,,
-f1_retained_erng,29,1994,TOTAL Dividends Declared-Preferred Stock (Acct. 437),,,,
-f1_retained_erng,30,1994,Dividends Declared-Common Stock (Account 438),,,,
-f1_retained_erng,31,1994,Line 31,,,,
-f1_retained_erng,32,1994,Line 32,,,,
-f1_retained_erng,33,1994,Line 33,,,,
-f1_retained_erng,34,1994,Line 34,,,,
-f1_retained_erng,35,1994,Line 35,,,,
-f1_retained_erng,36,1994,TOTAL Dividends Declared-Common Stock (Acct. 438),,,,
-f1_retained_erng,37,1994,"Transfers from Acct 216.1, Unapprop. Undistrib. Subsidiary Earnings",,,,
-f1_retained_erng,38,1994,"Balance - End of Year (Total 1,9,15,16,22,29,36,37)",,,,
-f1_retained_erng,39,1994,Line 39,,,,
-f1_retained_erng,40,1994,Line 40,,,,
-f1_retained_erng,41,1994,Line 41,,,,
-f1_retained_erng,42,1994,Line 42,,,,
-f1_retained_erng,43,1994,Line 43,,,,
-f1_retained_erng,44,1994,Line 44,,,,
-f1_retained_erng,45,1994,TOTAL Appropriated Retained Earnings (Account 215),,,,
-f1_retained_erng,46,1994,"TOTAL Approp. Retained Earnings-Amort. Reserve, Federal (Acct. 215.1)",,,,
-f1_retained_erng,47,1994,"TOTAL Approp. Retained Earnings (Acct. 215, 215.1) (Total 45,46)",,,,
-f1_retained_erng,48,1994,"TOTAL Retained Earnings (Account 215, 215.1, 216) (Total 38, 47)",,,,
-f1_retained_erng,49,1994,Balance-Beginning of Year (Debit or Credit),,,,
-f1_retained_erng,50,1994,Equity in Earnings for Year (Credit) (Account 418.1),,,,
-f1_retained_erng,51,1994,(Less) Dividends Received (Debit),,,,
-f1_retained_erng,52,1994,Other Changes (Explain),,,,
-f1_retained_erng,53,1994,Balance-End of Year (Total lines 49 thru 52),,,,
-f1_retained_erng,1,2004,Balance-Beginning of Period,,,,
-f1_retained_erng,38,2004,"Balance - End of Period (Total 1,9,15,16,22,29,36,37)",,,,
-f1_retained_erng,48,2004,"TOTAL Retained Earnings (Acct. 215, 215.1, 216) (Total 38, 47) (216.1)",,,,
+f1_retained_erng,1,1994,Balance-Beginning of Year,reported_value,unappropriated_retained_earnings_previous_year,,
```

Comment:
a few of these factiods have `_previous_year` or `_current_year` suffixes. these are dealt with in the `condense_double_year_earnings_types_dbf` transformer

Reply:
Hmm, I'm not I feel great about this -- one of the validations of this file that we would previously have been able to do to guard against manual errors is checking that each (`report_year`, `row_number`) combination has a unique `xbrl_factoid` associated with it, which this particular mapping breaks.

Reply:
They are unique! there are two `unappropriated_retained_earnings_previous_year` but the second is from 2004.

If we take away the double-year reporting both row 1 and 38 should be `unappropriated_retained_earnings`, but now row 1 is `unappropriated_retained_earnings_previous_year` and row 38 is `unappropriated_retained_earnings_current_year`